### PR TITLE
DOC: fixing overlapping text and wrong example

### PR DESCRIPTION
**Pull request by aabadie**
I found this overlapping text in the online documentation (see https://nilearn.github.io/manipulating_visualizing/manipulating_images.html#fetching-open-datasets). I took this opportunity to fix the full dataset example.

### CODE REVIEW

File: doc/manipulating_visualizing/manipulating_images.rst [hunk 1]
```diff
@@ -32,21 +32,25 @@ datasets and atlases. Dataset fetching functions can be imported from
 :mod:`nilearn.datasets`::
 
     >>> from nilearn import datasets
-    >>> haxby_files = datasets.fetch_haxby(n_subjects=1)  # doctest: +SKIP
+    >>> haxby_dataset = datasets.fetch_haxby(n_subjects=1)
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: Was removing the doctest skip intentional? This will trigger a download of the data while doctesting, which is probably something that we want to avoid.

File: doc/manipulating_visualizing/manipulating_images.rst [hunk 2]
```diff
@@ -32,21 +32,25 @@ datasets and atlases. Dataset fetching functions can be imported from
...
 
     >>> # The different files
-    >>> print(sorted(list(haxby_files.keys())))  # doctest: +SKIP
-    ['anat', 'description', 'func', 'mask_face', 'mask_face_little', 'mask_house', 'mask_house_little', 'mask_vt', 'session_target']
+    >>> print(sorted(list(haxby_dataset.keys()))) # doctest: +SKIP
+    ['anat', 'description', 'func', 'mask_face', 'mask_face_little',
+    'mask_house', 'mask_house_little', 'mask_vt', 'session_target']
     >>> # Path to first functional file
-    >>> print(haxby_files.func[0])  # doctest: +ELLIPSIS +SKIP
+    >>> print(haxby_dataset.func[0]) # doctest: +ELLIPSIS
```
Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: Same remark about skipping the doctest. It's probably a bad idea not to skip them, as it will entail heavy downloads on travis.

Another option would be to mock or modify dataset downloads while testing. I'm open to it, but it requires much more work.

**Comment by GaelVaroquaux** [comment]
In general the changes look right, but I don't think that we should be running downloads during doctests.

File: doc/manipulating_visualizing/manipulating_images.rst (hunk 1 above)
Comment (aabadie) [inline doc/manipulating_visualizing/manipulating_images.rst]: > Was removing the doctest skip intentional?

Yes, I just wanted to be sure the example was valid when running `make test-doc`. I can revert this change.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: > Yes, I just wanted to be sure the example was valid when running make
> test-doc.

Make sense. I wonder if we can have multiple modes. It probably goes in
the direction of mocking the downloaders. @AlexandreAbraham might have an
idea on this.

> I can revert this change.

It would be good.

Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/manipulating_images.rst]: Depending on which level you want to mock the downloader, it may not be as easy as it seems.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: > Depending on which level you want to mock the downloader, it may not be
> as easy as it seems.

I was thinking on a very high level: returning meaningful paths without
any download. The next step would even be to put some generated data in
these meaningful paths.

**Comment by aabadie** [comment]
> In general the changes look right, but I don't think that we should be running downloads during doctests.

I pushed the change. Waiting for CI's.

File: doc/manipulating_visualizing/manipulating_images.rst (hunk 1 above)
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/manipulating_images.rst]: Returning meaningful paths is already what we do in the tests. We can easily generate data in them. However, we sometimes have to generate file with meaningful content (XML) and this is the most difficult part to handle. I mean that we can't automatically mock everything. Some dataset loader requires specific code and we should think of a good way to put that out of the tests  in order to use it elsewhere.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: Are we already returning paths without downloading the doctests? If so we
shouldn't need to skip that part?

Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/manipulating_images.rst]: If you call `nilearn.datasets.tests.test_utils.setup_mock`, the function `fetch_haxby` will return existing paths to empty files.

Comment (GaelVaroquaux) [inline doc/manipulating_visualizing/manipulating_images.rst]: Alex Abadie: do you want to try to set this up, and thus remove the
skips?

Comment (aabadie) [inline doc/manipulating_visualizing/manipulating_images.rst]: Sure :)

File: doc/manipulating_visualizing/manipulating_images.rst [hunk 3]
```diff
@@ -32,21 +32,25 @@ datasets and atlases. Dataset fetching functions can be imported from
 :mod:`nilearn.datasets`::
 
     >>> from nilearn import datasets
-    >>> haxby_files = datasets.fetch_haxby(n_subjects=1)  # doctest: +SKIP
+    >>> haxby_dataset = datasets.fetch_haxby(n_subjects=1) # doctest: +SKIP
```
Comment (lesteve) [inline doc/manipulating_visualizing/manipulating_images.rst]: PEP8 two spaces before an inline comment. Just saying.

File: doc/manipulating_visualizing/manipulating_images.rst [hunk 4]
```diff
@@ -32,21 +32,25 @@ datasets and atlases. Dataset fetching functions can be imported from
...
     /.../nilearn_data/haxby2001/subj1/bold.nii.gz
-    >>> # Provide information on the dataset
-    >>> print(haxby_files.description) # doctest: +ELLIPSIS +SKIP
-    Haxby 2001 results
+
+Explanation and further resources of the dataset at hand can be retrieved as
+follows::
+
+    >>> print(haxby_dataset.description) # doctest: +ELLIPSIS +SKIP
+    Haxby 2001 results...
```
Comment (lesteve) [inline doc/manipulating_visualizing/manipulating_images.rst]: I would remove the three dots here

**Comment by lesteve** [comment]
Could the text overlap not be fixed at the HTML level rather than in the rst, e.g. by setting overflow-x on the pre class in the CSS ? Not a CSS expert so there might be a better way to do it.

nilearn_text_overlap

**Comment by GaelVaroquaux** [comment]
> Could the text overlap not be fixed at the HTML level rather than in
> the rst, e.g. by setting overflow-x on the pre class in the CSS ? Not a
> CSS expert so there might be a better way to do it.

Any change on CSS needs to be validated thoroughly, as it affects the
whole documentation. These things are tricky.

Backporting some CSS from scipy-lectures to nilearn would probably fix
this, as a lot of effort went into designing a very good CSS for the code
block in scipy-lectures. But this is a certain amount of work.

**Comment by lesteve** [comment]
> Any change on CSS needs to be validated thoroughly, as it affects the whole documentation. These things are tricky. 

Makes sense.

File: doc/manipulating_visualizing/manipulating_images.rst (hunk 1 above)
Comment (aabadie) [inline doc/manipulating_visualizing/manipulating_images.rst]: @GaelVaroquaux @AlexandreAbraham, I gave a try to `setup_mock` but it doesn't work like this out of the box and would require a bit more work (I cannot evaluate how many).
I reverted the change and now skip the doctest for this. The mock thing could be part of another PR I think.

**Comment by aabadie** [comment]
@lesteve, @GaelVaroquaux, for the record about the overlap issue, using `# doctest: +NORMALIZE_WHITESPACE` works with the output on 2 lines.

**Comment by lesteve** [comment]
LGTM, I'll wait for CircleCI green light, and I'll merge this one.

**Comment by lesteve** [comment]
Merging, thanks !
